src/monodroid/jni/internal-pinvoke-api.cc

```diff
@@ -0,0 +1,326 @@
+_monodroid_gref_log (const char *message)
+{
+	internal_calls->monodroid_gref_log (message);
+}
+
+MONO_API int
+_monodroid_gref_log_new (jobject curHandle, char curType, jobject newHandle, char newType, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	return internal_calls->monodroid_gref_log_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_gref_log_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_gref_log_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_weak_gref_new (jobject curHandle, char curType, jobject newHandle, char newType, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_weak_gref_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_weak_gref_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_weak_gref_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_lref_log_new (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_lref_log_new (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_lref_log_delete (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_lref_log_delete (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_gc_wait_for_bridge_processing (void)
+{
+	internal_calls->monodroid_gc_wait_for_bridge_processing ();
+}
+
+/* !DO NOT REMOVE! Used by Mono BCL */
+MONO_API int
+_monodroid_get_android_api_level (void)
+{
+	return internal_calls->monodroid_get_android_api_level ();
+}
+
+/* Can be called by a native debugger to break the wait on startup */
+MONO_API void
+monodroid_clear_gdb_wait (void)
+{
+	internal_calls->monodroid_clear_gdb_wait ();
+}
+
+MONO_API void*
+_monodroid_get_identity_hash_code (JNIEnv *env, void *v)
+{
+	return internal_calls->monodroid_get_identity_hash_code (env, v);
+}
+
+MONO_API void*
+_monodroid_timezone_get_default_id (void)
+{
+	return internal_calls->monodroid_timezone_get_default_id ();
+}
+
+MONO_API void
+_monodroid_counters_dump (const char *format, ...)
+{
+	va_list args;
+	va_start (args, format);
+	internal_calls->dump_counters (format, args);
+	va_end (args);
+}
+
+MONO_API int
+monodroid_embedded_assemblies_set_assemblies_prefix (const char *prefix)
+{
+	return internal_calls->monodroid_embedded_assemblies_set_assemblies_prefix (prefix);
+}
+
+MONO_API managed_timing_sequence*
+monodroid_timing_start (const char *message)
```

Comment:
I find it somewhat amusing that some symbols start with `_monodroid`, and some start with `monodroid`, and I *think* the idea was that `_monodroid` was for "things internal to `Mono.Android.dll`."

By that logic, I think this should be `_monodroid`?  Or we just ignore this "weird" difference?

Reply:
I don't like the leading underscore, it makes for weird looking invocations like, in this case, `JNIEnv._monodroid_timing_start` - it looks just... weird